### PR TITLE
Add highlight behavior

### PR DESCRIPTION
**Pull request by mbabker**
This pull request extracts the highlight JHtml behavior and associated media from Finder for use in the platform.

### CODE REVIEW

**Comment by eddieajau** [comment]
Hi Michael.  Is this to help with the Finder integration?

**Comment by mbabker** [comment]
This one is in part to help with integrating Finder and additionally
making the highlight behavior available to other components without
needing to duplicate code.  On the Finder repo, I've started working on
converting the system plugin that handled it previously into a content
plugin acting onContentPrepare which could be offered into the CMS before
Finder if completed and working OK.

On 9/21/11 2:32 AM, "Andrew Eddie"
reply@reply.github.com
 wrote:

**Comment by jlover** [comment]
**Checkstyle error details:**
libraries/joomla/html/html/string.php:179
Concat operator must be surrounded by spaces
libraries/joomla/html/html/string.php:179
Concat operator must be surrounded by spaces

File: libraries/joomla/html/html/string.php [hunk 1]
```diff
@@ -151,4 +151,30 @@ abstract class JHtmlString
...
+	{
+		// Get the document object.
+		$doc = JFactory::getDocument();
+
+		// We only want to highlight text on regular html pages.
+		if ($doc->getType() == 'html' && JFactory::getApplication()->input->get('tmpl', null, 'cmd') !== 'component')
+		{
+			// Add the highlighter media.
+			$uncompressed = JFactory::getConfig()->get('debug') ? '-uncompressed' : '';
+			JHtml::_('script', 'system/highlighter' . $uncompressed . '.js', true, true);
```
Comment (chdemko) [inline libraries/joomla/html/html/string.php]: These two lines can be replaced now by
    JHtml::_('script', 'system/highlighter.js', true, true);

(the -uncompressed is automatically added)

File: libraries/joomla/html/html/string.php [hunk 2]
```diff
@@ -151,4 +151,30 @@ abstract class JHtmlString
...
+	 *
+	 * @since   11.4
+	 */
+	public static function highlighter($terms)
+	{
+		// Get the document object.
+		$doc = JFactory::getDocument();
+
+		// We only want to highlight text on regular html pages.
+		if ($doc->getType() == 'html' && JFactory::getApplication()->input->get('tmpl', null, 'cmd') !== 'component')
```
Comment (chdemko) [inline libraries/joomla/html/html/string.php:170]: Is it to this function to test the doc type or the calling one?

Comment (mbabker) [inline libraries/joomla/html/html/string.php:170]: It should only trigger the behavior if the doc type being rendered is HTML and not using the component.php template file.

**Comment by mbabker** [comment]
Closing in favor of #608

**Pull request closed without merging**